iohub/dataset_base.py

```diff
@@ -0,0 +1,71 @@
+from abc import ABC, abstractmethod
+from typing import Generator, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+
+class BaseFOV(ABC):
+    # NOTE: not using the `data` method from ngff Position
+
+    @property
+    @abstractmethod
+    def channels(self) -> list[str]:
+        raise NotImplementedError
+
+    def channel_index(self, key: str) -> int:
+        """Return index of given channel."""
+        return self.channels.index(key)
+
+    @property
+    @abstractmethod
+    def shape(self) -> tuple[int, int, int, int, int]:
+        # NOTE: suggestion, fix dimension to 5?
+        # We could me more restrictive than ome-zarr
+        raise NotImplementedError
+
+    @abstractmethod
+    def __getitem__(
+        self,
+        key: Union[int, slice, tuple[Union[int, slice], ...]],
+    ) -> ArrayLike:
+        """
+        Output object must support __array__ interface, np.asarray(...).
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def ndim(self) -> int:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def dtype(self) -> np.dtype:
+        raise NotImplementedError
+
+    @property
+    def scale(self) -> list[float]:
+        """Helper function for FOV scale."""
+        raise NotImplementedError
+
+
+class BaseFOVCollection(ABC):
```

Comment:
```suggestion
class BaseFOVCollection(ABC):
    @abstractmethod
    def __enter__(self) -> BaseFOVCollection:
        "Open the underlying file and return self."
        raise NotImplementedError

    @abstractmethod
    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
        "Close the files."
        raise NotImplementedError
```

Can we also have context manager here?